rskj-core/src/main/java/co/rsk/peg/union/UnionBridgeSupportImpl.java

```diff
@@ -146,6 +149,60 @@ public UnionResponseCode increaseLockingCap(Transaction tx, Coin newCap) {
         return UnionResponseCode.SUCCESS;
     }
 
+    @Override
+    public UnionResponseCode requestUnionRbtc(Transaction tx, co.rsk.core.Coin amount) {
+        final String REQUEST_UNION_RBTC_TAG = "requestUnionRbtc";
+
+        if (isCallerNoUnionBridgeContractAddress(tx)) {
+            return UnionResponseCode.UNAUTHORIZED_CALLER;
+        }
+
+        if (isInvalidAmount(amount)) {
+            return UnionResponseCode.INVALID_VALUE;
+        }
+
+        storageProvider.setWeisTransferredToUnionBridge(amount);
+        logger.info("[{}] Amount requested by the union bridge has been transferred. Amount Requested: {}.", REQUEST_UNION_RBTC_TAG, amount);
+        return UnionResponseCode.SUCCESS;
+    }
+
+    private boolean isCallerNoUnionBridgeContractAddress(Transaction tx) {
+        RskAddress unionBridgeContractAddress = getUnionBridgeContractAddress();
+        boolean isCallerNoUnionBridgeContractAddress = !tx.getSender(signatureCache).equals(unionBridgeContractAddress);
+        if (isCallerNoUnionBridgeContractAddress) {
+            String baseMessage = String.format("Caller is not the Union Bridge Contract Address. Caller address: %s, Union Bridge Contract Address: %s", tx.getSender(), unionBridgeContractAddress);
+            logger.warn(LOG_PATTERN, "isCallerNoUnionBridgeContractAddress", baseMessage);
+        }
+        return isCallerNoUnionBridgeContractAddress;
+    }
+
+    private boolean isInvalidAmount(co.rsk.core.Coin amountRequested) {
+        boolean isAmountNullOrGreaterThanZero = amountRequested == null || amountRequested.compareTo(co.rsk.core.Coin.ZERO) < 1;
+        if (isAmountNullOrGreaterThanZero) {
```

Comment:
So is a zero value allowed or not? Log message says one thing, code says another. Is there a test?